# Assignment-3(Hypothesis Testing)
Q.1) A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.
Solution: 
Step 1 - 
H0: there is no change in diameter
Ha:there is change in diameter
if p-value is > 0.05 = Accept null hypothesis
if p-value is < 0.05 = Reject null hypothesis
Alternate hypothesis is used when we take action or p-value is less than 0.05
Step 2 - Identify test statistic used for testing the validity of null hypothesis - t test
Step 3 - Decide Significance value(alpha)=0.05
Step 4 - Calculate p value = 0.472
Step 5 - Compare p_value with α = 0.05 (At 5% significance level)

In [213]:
import numpy as np
from scipy import stats
import pandas as pd

In [214]:
data = pd.read_csv("Cutlets.csv")

In [215]:
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [216]:
data.tail()

Unit A  Unit B
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [217]:
Unit_A=data['Unit A'].mean()
Unit_A

7.01909142857143

In [218]:
Unit_B=data['Unit B'].mean()
Unit_B

6.964297142857142

In [219]:
print('Unit A Mean > Unit B Mean = ',Unit_A>Unit_B)

Unit A Mean > Unit B Mean =  True


In [220]:
UA=data['Unit A']

In [221]:
UA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [222]:
UB=data['Unit B']

In [223]:
UB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [224]:
stats.ttest_ind( UA,UB)

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [225]:
# 2-sample 2-tail ttest:
p_value=stats.ttest_ind( UA,UB)[1]
p_value                               # 2-tail probability 

0.4722394724599501

In [226]:
# compare p_value with α = 0.05 (At 5% significance level)
if p_value < 0.05:
  print('Reject null hypothesis')
else:
  print('Accept null hypothesis')

Accept null hypothesis


Inference - There is no significant difference in the diameters of both Unit A and Unit B.

Q.2) A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. 
Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.


Hypothesis:

There are no significant differences between the groups' mean Lab values.
H0:μ1=μ2=μ3=μ4=μ5
    
There is a significant difference between the groups' mean Lab values.
H1:μ1≠μ2≠μ3≠μ4

In [227]:
data1=pd.read_csv('LabTAT.csv')
data1.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [228]:
Lab_1=data1['Laboratory 1'].mean()
Lab_1

178.36158333333339

In [229]:
Lab_2=data1['Laboratory 2'].mean()
Lab_2

178.9029166666668

In [230]:
Lab_3=data1['Laboratory 3'].mean()
Lab_3

199.91325000000003

In [231]:
Lab_4=data1['Laboratory 4'].mean()
Lab_4

163.6827499999999

In [232]:
import scipy.stats as stats
p_value1=stats.f_oneway(data1.iloc[:,0], data1.iloc[:,1],data1.iloc[:,2],data1.iloc[:,3])
p_value1

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [233]:
p_value1

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Compare p_value1 with alpha=0.05
P_value is 0.00 < 0.05= Accept H1, hence Average of at least 1 laboratory are different.

Q.3)  Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.
	   East	 West North	South
Males	50	 142	131	 70
Females	550	 351	480	 350
H0  : All proportions are equal
H1  : Not all Proportions are equal
1. Check p-value
2. If p-Value < alpha, we reject Null Hypothesis

In [234]:
data2=pd.read_csv('BuyerRatio.csv')
data2.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [235]:
Male = [50,142,131,70]
Female=[435,1523,1356,750]
Sales=[Male,Female]
print(Sales)

[[50, 142, 131, 70], [435, 1523, 1356, 750]]


In [236]:
chiStats = sp.stats.chi2_contingency(Sales)
print('Test t=%f p-value=%f' % (chiStats[0], chiStats[1]))

Test t=1.595946 p-value=0.660309


In [237]:
if chiStats[1] < 0.05:
  print('we reject null hypothesis')
else:
  print('we accept null hypothesis')

we accept null hypothesis


By P-Value=0.66 > 0.05 hence we accept null hypothesis

In [238]:
alpha = 0.05
critical_value = sp.stats.chi2.ppf(q = 1 - alpha,df=chiStats[2])

In [239]:
critical_value

7.814727903251179

In [240]:
observed_chi_val = chiStats[0]
observed_chi_val

1.595945538661058

observed_chi_val < critical_value , 
1.59 < 7.81
Hence, We cannot reject null hypothesis

Q.4) TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


In [197]:
Customer = pd.read_csv('Costomer+OrderForm.csv')
Customer.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [241]:
Phillippines_value=Customer['Phillippines'].value_counts()
Phillippines_value

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [242]:
Indonesia_value=Customer['Indonesia'].value_counts()
Indonesia_value

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [243]:
Malta_value=Customer['Malta'].value_counts()
Malta_value

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [244]:
India_value=Customer['India'].value_counts()
India_value

Error Free    280
Defective      20
Name: India, dtype: int64

In [245]:
chiStats = sp.stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])
print('Test t=%f p-value=%f' % (chiStats[0], chiStats[1]))

Test t=3.858961 p-value=0.277102


In [246]:
if chiStats[1] < 0.05:
  print('we reject null hypothesis')
else:
  print('we accept null hypothesis')

we accept null hypothesis


By P-Value=0.66 > 0.05 hence we accept null hypothesis

In [247]:
alpha = 0.05
critical_value = sp.stats.chi2.ppf(q = 1 - alpha,df=chiStats[2])

In [248]:
critical_value

7.814727903251179

In [249]:
observed_chi_val = chiStats[0]
observed_chi_val

3.858960685820355

observed_chi_val < critical_value , 
3.85 < 7.81
Hence, We cannot reject null hypothesis

Inference is that proportion of defective % across the center is same.